src/offset/local/tz_info/timezone.rs

```diff
@@ -629,6 +640,63 @@ fn find_tz_file(path: impl AsRef<Path>) -> Result<File, Error> {
     }
 }
 
+#[cfg(target_env = "ohos")]
+const TZDATA_PATH: &str = "/system/etc/zoneinfo/tzdata";
+
+#[cfg(target_env = "ohos")]
+fn from_tzdata_bytes(bytes: &mut Vec<u8>, tz_string: &str) -> Result<Vec<u8>, Error> {
+    const VERSION_SIZE: usize = 12;
+    const OFFSET_SIZE: usize = 4;
+    const INDEX_CHUNK_SIZE: usize = 48;
+    const ZONENAME_SIZE: usize = 40;
+
+    let mut cursor = Cursor::new(&bytes);
+    // version head
+    let _ = cursor.read_exact(VERSION_SIZE)?;
+    let index_offset_offset = cursor.read_be_u32()?;
+    let data_offset_offset = cursor.read_be_u32()?;
+    // final offset
+    let _ = cursor.read_be_u32()?;
+
+    if cursor.seek_after(index_offset_offset as usize).is_err() {
+        return Err(Error::Io(io::ErrorKind::NotFound.into()));
+    }
+    let mut idx = index_offset_offset;
+    while idx < data_offset_offset {
+        let index_buf = cursor.read_exact(ZONENAME_SIZE)?;
+        let offset = cursor.read_be_u32()?;
+        let length = cursor.read_be_u32()?;
+        let zone_name = str::from_utf8(index_buf)?.trim_end_matches('\0');
+        if zone_name == tz_string {
+            if cursor.seek_after((data_offset_offset + offset) as usize).is_err() {
+                return Err(Error::Io(io::ErrorKind::NotFound.into()));
+            }
+            return match cursor.read_exact(length as usize) {
+                Ok(result) => Ok(result.to_vec()),
+                _ => Err(Error::Io(io::ErrorKind::NotFound.into())),
+            };
+        }
+        idx += INDEX_CHUNK_SIZE as u32;
+    }
+
+    Err(Error::Io(io::ErrorKind::NotFound.into()))
+}
+
+#[cfg(target_env = "ohos")]
+fn from_tzdata_file(file: &mut File, tz_string: &str) -> Result<Vec<u8>, Error> {
+    let mut bytes: Vec<u8> = Vec::new();
+    file.read_to_end(&mut bytes)?;
+    from_tzdata_bytes(&mut bytes, tz_string)
+}
+
+#[cfg(target_env = "ohos")]
+fn find_ohos_tz_data(tz_string: &str) -> Result<Vec<u8>, Error> {
+    if let Ok(mut file) = File::open(TZDATA_PATH) {
+        return from_tzdata_file(&mut file, tz_string);
+    }
+    Err(Error::Io(io::ErrorKind::NotFound.into()))
```

Comment:
There doesn't seem to be much point in synthesizing an `io::Error` here if we can just use the one from `File::open()`.